### ISSUE TITLE
crash assembling some invalid commands

### ISSUE THREAD

**Issue by psifertex** [Vector35/binaryninja-api#2361]
May be arm specific. 

1. Download https://github.com/JonathanSalwan/binary-samples/blob/master/elf-Linux-ARMv7-ls
2. Patch offset 0x9ee8 with:
```ldr r1, data_ad00```
(note the missing 0x from the original)

Crashes.

Expected behavior: invalid instruction warning because relative patches and symbols not supported.

**Comment by lwerdna** [comment]
Jeez lldb's not too helpful either... 

```
...
2021-04-09 21:44:49.267228-0400 binaryninja[38109:3107778] GVA info: preferred scaler idx 1
2021-04-09 21:44:53.093880-0400 binaryninja[38109:3107708] SecTaskLoadEntitlements failed error=22 cs_flags=20, pid=38109
2021-04-09 21:44:53.093950-0400 binaryninja[38109:3107708] SecTaskCopyDebugDescription: binaryninja[38109]/0#-1 LF=0
Process 38109 exited with status = 1 (0x00000001)
```

I wonder if it's some terrible exception path in llvm that ends exit() or something :/

**Comment by lwerdna** [comment]
Hehe it really is "some terrible exception path in llvm that ends exit() or something"!

[backtrace.txt](https://github.com/Vector35/binaryninja-api/files/6289198/backtrace.txt)

What a way to fail when assembly goes wrong... just terminate the whole process. Looking into it, but it stinks being at the mercy of this llvm blob. (thinking also of the assemble-at-address issue)

**Comment by fuzyll** [comment]
This still repros for me on 3.0.3321-dev.

**Comment by negasora** [comment]
This was fixed before 3.1.3545-dev, now it shows an error dialog
